src/main/kotlin/org/rust/lang/core/types/ty/Ty.kt

```diff
@@ -160,7 +160,7 @@ fun Ty.isMovesByDefault(lookup: ImplLookup): Boolean =
     when (this) {
         is TyUnknown, is TyReference, is TyPointer, is TyFunction -> false
         is TyTuple -> types.any { it.isMovesByDefault(lookup) }
-        is TyTypeParameter -> parameter != TyTypeParameter.Self
+        is TyTypeParameter -> parameter != TyTypeParameter.Self && lookup.isCopy(this).not()
```

Comment:
`not()` method for boolean value looks quite weird

Reply:
I find `.not()` more visible than `!`. But if it looks weird, I'll replace it.

Reply:
An outsider's take here -- in most languages I've used, the `!` prefix is idiomatic. People read it as "not <expression>" -- it works out fine.